build.sbt

```diff
@@ -1,70 +1,78 @@
     url("https://github.com/DavidGregory084")
   )
 )
-homepage := scmInfo.value.map(_.browseUrl)
-
-crossSbtVersions := Seq("1.6.2")
-
-enablePlugins(SbtPlugin)
-
-// License headers
 
-Compile / headerCreate := { (Compile / headerCreate).triggeredBy(Compile / compile).value }
-Test / headerCreate    := { (Test / headerCreate).triggeredBy(Test / compile).value }
-
-scalacOptions += "-Xlint:unused"
-
-libraryDependencies ++= Seq(
-  "org.scalatest"     %% "scalatest"       % "3.2.11"   % Test,
-  "org.scalacheck"    %% "scalacheck"      % "1.15.4"   % Test,
-  "org.scalatestplus" %% "scalacheck-1-15" % "3.2.11.0" % Test
-)
+ThisBuild / homepage := scmInfo.value.map(_.browseUrl)
 
 ThisBuild / semanticdbEnabled                              := true
 ThisBuild / semanticdbVersion                              := scalafixSemanticdb.revision
 ThisBuild / scalafixDependencies += "com.github.liancheng" %% "organize-imports" % "0.6.0"
 
 ThisBuild / versionScheme := Some(VersionScheme.EarlySemVer)
 
-mimaPreviousArtifacts := Set(
-  projectID.value.withRevision("0.4.0")
-)
-
-mimaBinaryIssueFilters ++= Seq(
-)
-
-// Testing
-
-scriptedBufferLog := false
+ThisBuild / crossSbtVersions := Seq("1.6.2")
+
+lazy val `sbt-tpolecat` = project
+  .in(file("."))
+  .aggregate(`sbt-tpolecat-plugin`)
+  .settings(
+    publish                := {},
+    publishLocal           := {},
+    publishArtifact        := false,
+    publish / skip         := true,
+    mimaReportBinaryIssues := {}
+  )
 
-scriptedLaunchOpts := scriptedLaunchOpts.value ++ Seq(
-  "-Xmx1024M",
-  "-Dplugin.version=" + version.value
-)
+lazy val `sbt-tpolecat-plugin` = project
+  .in(file("plugin"))
+  .enablePlugins(SbtPlugin)
+  .settings(
+    name                   := "sbt-tpolecat",
+    moduleName             := "sbt-tpolecat",
+    description            := "scalac options for the enlightened",
+    Compile / headerCreate := { (Compile / headerCreate).triggeredBy(Compile / compile).value },
+    Test / headerCreate    := { (Test / headerCreate).triggeredBy(Test / compile).value },
+    scalacOptions += "-Xlint:unused",
+    libraryDependencies ++= Seq(
+      "org.scalatest"     %% "scalatest"       % "3.2.11"   % Test,
+      "org.scalacheck"    %% "scalacheck"      % "1.15.4"   % Test,
+      "org.scalatestplus" %% "scalacheck-1-15" % "3.2.11.0" % Test
+    ),
+    mimaPreviousArtifacts := Set(
+      projectID.value.withRevision("0.4.0")
+    ),
+    mimaBinaryIssueFilters ++= Seq(
+    ),
+    scriptedBufferLog := false,
+    scriptedLaunchOpts := scriptedLaunchOpts.value ++ Seq(
+      "-Xmx1024M",
+      "-Dplugin.version=" + version.value
+    ),
+    test := {
+      (Test / test).value
+      scripted.toTask("").value
+    }
+  )
 
-test := {
-  (Test / test).value
-  scripted.toTask("").value
-}
+lazy val `sbt-tpolecat-scalafix` = scalafixProject("sbt-tpolecat")
+  .inputSettings(
+    libraryDependencies += (`sbt-tpolecat-plugin` / projectID).value.withRevision("0.4.0")
+  )
```

Comment:
Ah, I see what you did here :) I've wondered if it was possible, most projects don't seem to publish their scalafixes but keep them under a separate directory in the repo.